cmd/fdsn-ws/fdsn_station.go

```diff
@@ -663,14 +665,20 @@ func (n *NetworkType) doFilter(params []fdsnStationV1Search) bool {
 
 	if len(n.Station) == 0 {
 		// for network nodes without children (though unlikely to happen):
-		//   1. If the query parameter stops at network level, then the match is done
-		//   2. Otherwise, we're unable to get a match because of empty children (thus returning false)
+		//   If there are query parameters for furthur levels, and there's no "*" (match anything) in the parameters,
+		//     then it'll be impossible to find a match (because we don't have children)
 		for _, p := range matchedParams {
-			if p.StationReg == nil && p.ChannelReg == nil && p.LocationReg == nil {
-				return true
+			if p.StationReg != nil && !contains(p.StationReg, REGEX_ANYTHING) {
```

Comment:
What does the regex check add?

Reply:
See Jerome's comment regarding issue with WTSZ here https://github.com/GeoNet/tickets/issues/12671#issuecomment-1452655370